### PR TITLE
Add reposted + purchased playlists to library C-2951

### PR DESCRIPTION
**Pull request by nicoback2**
### Description
To be merged into feature branch!

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/common/src/api/library.ts [hunk 1]
```diff
@@ -9,66 +9,96 @@ import { reformatCollection } from 'store/cache/collections/utils/reformatCollec
...
+  }
+  const { data: rawCollections = [] } =
+    collectionType === 'album'
+      ? await sdk.full.users.getUserLibraryAlbums(getCollectionsParams)
+      : await sdk.full.users.getUserLibraryPlaylists(getCollectionsParams)
+  const collectionsMetadata = rawCollections
+    .map((r: APIActivityV2) => makeActivity(r))
+    .filter(removeNullable) as UserCollectionMetadata[]
+  const collections = collectionsMetadata.map((am) =>
+    reformatCollection({
```
Comment (nicoback2) [inline packages/common/src/api/library.ts:67]: FYI need to incorporate suggestion to move this from last PR still!

Comment (amendelsohn) [reply packages/common/src/api/library.ts:67]: Awesome! Should be easy, just copy the pattern in createApi for `reformatUser`

File: packages/common/src/api/library.ts [hunk 2]
```diff
@@ -9,66 +9,96 @@ import { reformatCollection } from 'store/cache/collections/utils/reformatCollec
...
   offset: number
   limit: number
   query?: string
   sortMethod?: full.GetUserLibraryAlbumsSortMethodEnum
   sortDirection?: full.GetUserLibraryAlbumsSortDirectionEnum
 }
+const COLLECTIONS_CACHE_OPTIONS = {
+  kind: Kind.COLLECTIONS,
+  schemaKey: 'collections',
+  idListArgKey: 'playlist_id'
```
Comment (amendelsohn) [inline packages/common/src/api/library.ts:23]: I don't think the `idListArgKey` does anything here. This is only useful when you're fetching a list of known playlist ids

File: packages/web/src/pages/saved-page/components/desktop/PlaylistsTabPage.tsx [hunk 3]
```diff
@@ -69,11 +63,13 @@ export const PlaylistsTabPage = () => {
     )
   }, [dispatch])
 
-  const noSavedPlaylists =
-    accountPlaylistsStatus === Status.SUCCESS && savedAlbumIds.length === 0
-  const noFetchedResults = !statusIsNotFinalized(status) && cards.length === 0
+  if (statusIsNotFinalized(status)) {
+    // TODO(nkang) - Confirm loading state UI
```
Comment (amendelsohn) [inline packages/web/src/pages/saved-page/components/desktop/PlaylistsTabPage.tsx:67]: todo reminder

File: packages/web/src/pages/saved-page/components/desktop/PlaylistsTabPage.tsx [hunk 4]
```diff
@@ -29,35 +30,28 @@ const messages = {
...
+    {
+      pageSize: 20,
+      disabled: currentUserId == null
+    }
+  )
+
+  const noFetchedResults =
+    !statusIsNotFinalized(status) && fetchedPlaylists?.length === 0
+  const cards = fetchedPlaylists?.map(({ playlist_id: id }, i) => {
+    return <CollectionCard index={i} key={id} albumId={id} />
```
Comment (amendelsohn) [inline packages/web/src/pages/saved-page/components/desktop/PlaylistsTabPage.tsx:54]: is it worth memoizing this?

**Review by amendelsohn — APPROVED** [review]
Looks great! Exciting to see the audius-query usages 😃

File: packages/web/src/pages/saved-page/components/desktop/PlaylistsTabPage.tsx (hunk 4 above)
Comment (nicoback2) [reply packages/web/src/pages/saved-page/components/desktop/PlaylistsTabPage.tsx:54]: i think so, will do

**Pull request merged**